Documenting secrets management SDK client for .NET (#5175)
dapr/docs: sdkdocs/dotnet/content/en/dotnet-sdk-docs/dotnet-secrets/dotnet-secretsclient-usage.md

Using the built-in extension methods for registering the `DaprSecretsManagementClient` in a dependency injection
container can provide the benefit of registering the long-lived service a single time, centralize complex configuration
and improve performance by ensuring similarly long-lived resources are re-purposed when possible (e.g. `HttpClient`
instances).

There are three overloads available to give the developer the greatest flexibility in configuring the client for their
scenario. Each of these will register the `IHttpClientFactory` on your behalf if not already registered, and configure
the `DaprSecretsManagementClientBuilder` to use it when creating the `HttpClient` instance in order to re-use the same
instance as much as possible and avoid socket exhaustion and other issues.

In the first approach, there's no configuration done by the developer and the `DaprSecretsManagementClient` is
configured with the default settings.

```cs
var builder = WebApplication.CreateBuilder(args);

builder.Services.AddDaprSecretsManagementClient(); //Registers the `DaprSecretsManagementClient` to be injected as needed
var app = builder.Build();
```

> Sometimes the developer will need to configure the created client using the various configuration options detailed

Reviewer: ```suggestion
Sometimes you need to configure the created client using the various configuration options detailed
```
